lib/spack/spack/test/install.py

```diff
@@ -58,6 +58,25 @@ def fake_fetchify(url, pkg):
 
 
 @pytest.mark.usefixtures('install_mockery')
+def test_source_copy(mock_archive):
```

Comment:
Hmm and the `source_install=True` below should trigger the new code path or does the `mock_archive` doesn't have source?